testing/integration/ess/upgrade_broken_package_test.go

```diff
@@ -32,6 +32,7 @@ func TestUpgradeBrokenPackageVersion(t *testing.T) {
 		Local: false, // requires Agent installation
 		Sudo:  true,  // requires Agent installation
 	})
+	t.Skip("Skip this test because it is not compatible with .package-version pinning")
```

Comment:
This test cannot work when package version and the string contained in `version.go` diverge because of this [`snippet`](https://github.com/elastic/elastic-agent/blob/6c005a48168b9c3c15cde01bb5dc98396eec0fe5/internal/pkg/agent/application/paths/common.go#L218-L227) which will stop returning the correct versioned home.

Maybe we can find an automated way to disable the test when package version and the version string in code diverge (we can file a follow-up issue, we don't need to solve this in this PR)

Reply:
Created https://github.com/elastic/elastic-agent/issues/9188 to follow-up